### PR TITLE
Add visualizations to status page

### PR DESCRIPTION
**Pull request by driazati**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55 Add visualizations to status page**

This adds both a view of current GitHub actions job status (# of queued, pending, and running jobs, with data from https://github.com/pytorch/test-infra/pull/16) and a heatmap that shows how job failures (based on the jobs from pytorch-master) correlate with each other. The first is useful to see when GitHub is backed up on queueing, and the second for identifying jobs that commonly fail together (meaning we can probably get rid of one).

`yarn install` added a lot of changes to the lockfile in addition to the `@toast-ui` package, I don't know enough about JS to know if that's normal or not though (maybe because react-scripts has a [bunch of dependencies](https://www.npmjs.com/package/react-scripts) that got updated?)

Screen Shot 2021-05-31 at 10 25 40 PM

### CODE REVIEW

**Comment by zhouzhuojie** [comment]
Nice charts!!

BTW, is it related to https://github.com/pytorch/test-infra/pull/16? If we have more bandwidth, I think we can also explore a stable metrics system like prometheus https://aws.amazon.com/prometheus/, it helps to collect all different kind of OSS CI metrics like cost, time duration, histogram. And we can label these with high cardinality.

File: src/App.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ import QueueDisplay from "./QueueDisplay.js";
 import BuildHistoryDisplay from "./BuildHistoryDisplay.js";
 import GitHubStatusDisplay from "./GitHubStatusDisplay.js";
 import PerfHistoryDisplay from "./PerfHistoryDisplay.js";
+import JobCorrelationHeapmap from "./JobCorrelationHeatmap.js";
```
Comment (janeyx99) [inline src/App.js:8]:
```suggestion
import JobCorrelationHeatmap from "./JobCorrelationHeatmap.js";
```
nit

**Comment by driazati** [comment]
> Nice charts!!
> 
> BTW, is it related to [pytorch/test-infra#16](https://github.com/pytorch/test-infra/pull/16)? If we have more bandwidth, I think we can also explore a stable metrics system like prometheus https://aws.amazon.com/prometheus/, it helps to collect all different kind of OSS CI metrics like cost, time duration, histogram. And we can label these with high cardinality.

Yup it relies on the data added from that lambda (which is already running), prometheus + grafana would be cool if not a little involved since right now everything can be statically hosted (but it's probably not that big of a deal to set up an EC2 machine to serve it).

File: src/App.js [hunk 2]
```diff
@@ -116,6 +118,8 @@ const App = () => (
 
 const Status = () => (
   <div>
+    <GitHubActionsDisplay />
+    <JobCorrelationHeapmap />
```
Comment (janeyx99) [inline src/App.js:122]:
```suggestion
    <JobCorrelationHeatmap />
```

File: src/GitHubActionsDisplay.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    this.state = {};
+  }
+  componentDidMount() {
+    this.update();
+  }
+  async update() {
+    const data = {
+      actions_statuses: JSON.parse(
+        await fetch(
+          "https://ossci-checks-status.s3.amazonaws.com/status.json",
```
Comment (janeyx99) [inline src/GitHubActionsDisplay.js:17]: just for my own understanding: is this S3 file very frequently updated?

**Review by janeyx99 — COMMENTED** [review]
The rest of your code seems good to me but I don't feel entirely qualified to review ts

**Comment by zhouzhuojie** [comment]
> > Nice charts!!
> > BTW, is it related to [pytorch/test-infra#16](https://github.com/pytorch/test-infra/pull/16)? If we have more bandwidth, I think we can also explore a stable metrics system like prometheus https://aws.amazon.com/prometheus/, it helps to collect all different kind of OSS CI metrics like cost, time duration, histogram. And we can label these with high cardinality.
> 
> Yup it relies on the data added from that lambda (which is already running), prometheus + grafana would be cool if not a little involved since right now everything can be statically hosted (but it's probably not that big of a deal to set up an EC2 machine to serve it).

The setup cost is a one-time thing, and it opens wide opportunities for the "better measurement for OSS" that people think it's nice to have when publicly available. Agreed that currently the stats can be statically managed, and we can introduce a full prometheus solution when the requirements are there.

**Comment by ezyang** [comment]
btw we used to have a grafana/prometheus deploy at https://monitoring.ci.ezyang.com/ (but it looks pretty busted right now, maybe it's just the SSL certs that need fixing). Configuration for it lives here: https://github.com/fairinternal/ossci-infra/tree/master/config/templates/prometheus

**Comment by ezyang** [comment]
One thing to be aware of: prometheus's default operating model as a *scrape* based telemetry service is not very good for CI jobs, which are ephemeral. You can make it work with stuff like pushgateway but it's definitely not the happy path.

File: src/GitHubActionsDisplay.js (hunk 3 above)
Comment (driazati) [reply src/GitHubActionsDisplay.js:17]: the lambda runs every few minutes: https://console.aws.amazon.com/lambda/home?region=us-east-1#/functions/github-checks-status-updater?tab=code

**Comment by zhouzhuojie** [comment]
> One thing to be aware of: prometheus's default operating model as a _scrape_ based telemetry service is not very good for CI jobs, which are ephemeral. You can make it work with stuff like pushgateway but it's definitely not the happy path.

Didn't notice we already have this grafana/prometheus setup, that's nice. pushgateway does require a service running to listen for push events, and yes, that's one more thing to maintain. 

Though for the use cases like https://github.com/pytorch/test-infra/pull/16, it's already a _scrape_ job running with cron spec, and there are some github_action_exporter (https://github.com/search?q=github+action+exporter) we can use.

**Review by malfet — APPROVED** [review]

**Pull request merged**